example_test.go

```diff
@@ -10,6 +10,131 @@ import (
+	// Output:
+	// ipv4.Is4() -> true
+	// ipv6.Is4() -> false
+	// ipEmpty.Is4() -> false
+	// ipv4MappedToIPv6.Is4() -> false
+}
+func ExampleIP_Is4in6() {
+	ipv4 := netaddr.MustParseIP("192.0.2.3")
+	ipv6 := netaddr.MustParseIP("2001:db8::68")
+	ipEmpty := netaddr.IP{}
+	ipv4MappedToIPv6 := netaddr.MustParseIP("::ffff:192.0.2.3")
+
+	fmt.Printf("ipv4.Is4in6() -> %v\n", ipv4.Is4in6())
+	fmt.Printf("ipv6.Is4in6() -> %v\n", ipv6.Is4in6())
+	fmt.Printf("ipEmpty.Is4in6() -> %v\n", ipEmpty.Is4in6())
+	fmt.Printf("ipv4MappedToIPv6.Is4in6() -> %v\n", ipv4MappedToIPv6.Is4in6())
+	// Output:
+	// ipv4.Is4in6() -> false
+	// ipv6.Is4in6() -> false
+	// ipEmpty.Is4in6() -> false
+	// ipv4MappedToIPv6.Is4in6() -> true
+}
+func ExampleIP_Is6() {
+	ipv4 := netaddr.MustParseIP("192.0.2.3")
+	ipv6 := netaddr.MustParseIP("2001:db8::68")
+	ipEmpty := netaddr.IP{}
+	ipv4MappedToIPv6 := netaddr.MustParseIP("::ffff:192.0.2.3")
+
+	fmt.Printf("ipv4.Is6() -> %v\n", ipv4.Is6())
+	fmt.Printf("ipv6.Is6() -> %v\n", ipv6.Is6())
+	fmt.Printf("ipEmpty.Is6() -> %v\n", ipEmpty.Is6())
+	fmt.Printf("ipv4MappedToIPv6.Is6() -> %v\n", ipv4MappedToIPv6.Is6())
+	// Output:
+	// ipv4.Is6() -> false
+	// ipv6.Is6() -> true
+	// ipEmpty.Is6() -> false
+	// ipv4MappedToIPv6.Is6() -> true
+}
+
+func ExampleIP_IsZero() {
+	ipEmpty := netaddr.IP{}
+	ipv4AllZeroes := netaddr.MustParseIP("0.0.0.0")
+	ipv6AllZeroes := netaddr.MustParseIP("::")
+
+	fmt.Printf("ipEmpty.IsZero() -> %v\n", ipEmpty.IsZero())
+	fmt.Printf("ipv4AllZeroes.IsZero() -> %v\n", ipv4AllZeroes.IsZero())
+	fmt.Printf("ipv6AllZeroes.IsZero() -> %v\n", ipv6AllZeroes.IsZero())
+	// Output:
+	// ipEmpty.IsZero() -> true
+	// ipv4AllZeroes.IsZero() -> false
+	// ipv6AllZeroes.IsZero() -> false
+}
+
+func ExampleIP_String() {
+	ipv4 := netaddr.MustParseIP("192.0.2.3")
+	ipv6 := netaddr.MustParseIP("2001:db8::68")
+	ipv4MappedToIPv6 := netaddr.MustParseIP("::ffff:192.0.2.3")
+
+	fmt.Printf("ipv4.String() -> %v\n", ipv4.String())
+	fmt.Printf("ipv6.String() -> %v\n", ipv6.String())
+	fmt.Printf("ipv4MappedToIPv6.String() -> %v\n", ipv4MappedToIPv6.String())
+	// Output:
+	// ipv4.String() -> 192.0.2.3
+	// ipv6.String() -> 2001:db8::68
+	// ipv4MappedToIPv6.String() -> ::ffff:c000:203
+}
+
+func ExampleIP_Unmap() {
+	ipv4 := netaddr.MustParseIP("192.0.2.3")
+	ipv6 := netaddr.MustParseIP("2001:db8::68")
+	ipv4MappedToIPv6 := netaddr.MustParseIP("::ffff:192.0.2.3")
+
+	fmt.Printf("ipv4.Unmap() -> %v\n", ipv4.Unmap())
+	fmt.Printf("ipv6.Unmap() -> %v\n", ipv6.Unmap())
+	fmt.Printf("ipv4MappedToIPv6.Unmap() -> %v\n", ipv4MappedToIPv6.Unmap())
+	// Output:
+	// ipv4.Unmap() -> 192.0.2.3
+	// ipv6.Unmap() -> 2001:db8::68
+	// ipv4MappedToIPv6.Unmap() -> 192.0.2.3
+}
+
+func ExampleIP_WithZone() {
+	ipv4 := netaddr.MustParseIP("192.0.2.3")
+	ipv6 := netaddr.MustParseIP("2001:db8::68")
+	ipv6Zoned := netaddr.MustParseIP("2001:db8::68%eth0")
+
+	fmt.Printf("ipv4.WithZone(\"newzone\") -> %v\n", ipv4.WithZone("newzone"))
```

Comment:
Maybe do

```go
fmt.Printf(`(%v).WithZone("newzone")` -> %v\n, ipv4, ipv4.WithZone("neozone"))
```

Then the output doesn't require you to refer back to what `ipv4` is.

Reply:
Agree the output should show before and after. Tempted to make this a table as well. I'll try some things and see what sticks.